x-pack/plugins/security_solution/public/management/components/endpoint_responder/status_action.tsx

```diff
@@ -127,50 +127,83 @@ export const EndpointStatusActionResult = memo<
     return <ResultComponent showAs="pending" />;
   }
 
-  return (
-    <ResultComponent showTitle={false}>
-      <EuiFlexGroup wrap={false} responsive={false}>
-        <EuiFlexItem grow={false}>
-          <EuiText size="s">
-            <FormattedMessage
-              id="xpack.securitySolution.endpointResponseActions.status.agentStatus"
-              defaultMessage="Agent status"
-            />
-          </EuiText>
-          <EndpointAgentAndIsolationStatus
-            status={endpointDetails.host_status}
-            isIsolated={Boolean(endpointDetails.metadata.Endpoint.state?.isolation)}
-            {...pendingIsolationActions}
-          />
-        </EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <EuiText size="s">
-            <FormattedMessage
-              id="xpack.securitySolution.endpointResponseActions.status.version"
-              defaultMessage="Version"
-            />
-          </EuiText>
-          <EuiText>{endpointDetails.metadata.agent.version}</EuiText>
-        </EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <EuiText size="s">
-            <FormattedMessage
-              id="xpack.securitySolution.endpointResponseActions.status.policyStatus"
-              defaultMessage="Policy status"
-            />
-          </EuiText>
-          <EndpointAppliedPolicyStatus
-            policyApplied={endpointDetails.metadata.Endpoint.policy.applied}
-          />
-        </EuiFlexItem>
-        <EuiFlexItem>
-          <EuiText size="s">
-            <FormattedMessage
-              id="xpack.securitySolution.endpointResponseActions.status.lastActive"
-              defaultMessage="Last active"
-            />
-          </EuiText>
-          <EuiText>
+  const statusDescriptionList = () => {
+    const agentStatus = () => {
+      let isolateStatus = '';
+
+      if (pendingIsolationActions.pendingIsolate > 0) {
+        isolateStatus = i18n.translate(
+          'xpack.securitySolution.endpointResponseActions.status.isolating',
+          {
+            defaultMessage: 'Isolating',
+          }
+        );
+      } else if (pendingIsolationActions.pendingUnIsolate > 0) {
+        isolateStatus = i18n.translate(
+          'xpack.securitySolution.endpointResponseActions.status.releasing',
+          {
+            defaultMessage: 'Releasing',
+          }
+        );
+      } else if (endpointDetails.metadata.Endpoint.state?.isolation) {
+        isolateStatus = i18n.translate(
+          'xpack.securitySolution.endpointResponseActions.status.isolated',
+          {
+            defaultMessage: 'Isolated',
+          }
+        );
+      }
+
+      return `${agentStatusText(endpointDetails.host_status)}${
+        isolateStatus.length > 0 ? ` - ${isolateStatus}` : ''
+      }`;
+    };
+
+    const statusData = [
```

Comment:
This has also a dynamic result, move it in a `useMemo` will avoid unnecessary code execution during renders.

Reply:
@dasansol92 if I wrap the containing function into the `useCallBack()` and the `deps` array contains the dynamic data, won't this be taken care of?  If I put this as `useMemo()` we'd have a hook in a hook.

Reply:
Ohh! You are totally right! Github diff hided me the `useCallback`. No need for `useMemo` here then. Apologies for the noise.